### PR TITLE
Query : PeerStats

### PR DESCRIPTION
**Pull request by parv0888**
## Purpose
Adds support for query `PeerStats`

## Changes
Adds support for query `PeerStats`

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance
By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    private final String nodeId;
+
+    /**
+     * Latency to the peer.
+     */
+    private final long latency;
+
+    /**
+     * Total packets received from the Peer.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java:22]:
```suggestion
     * Total packets received from the peer.
```

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    private final long packetsReceived;
+
+    /**
+     * Total packets sent to the Peer.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java:27]:
```suggestion
     * Total packets sent to the peer.
```

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStatistics.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    private final long averageBPSIn;
+
+    /**
+     * Average bytes per second going out of the node.
+     */
+    private final long averageBPSOut;
+
+    /**
+     * List of peers with their statistics.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStatistics.java:26]:
```suggestion
     * List of peers that the node is connected to containing their statistics.
```

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import concordium.ConcordiumP2PRpc;
+import lombok.Builder;
+import lombok.Getter;
+
+@Builder
+@Getter
+public class PeerStat {
+
+    /**
+     * Peer Node Id. Serialized has hex string.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java:12]:
```suggestion
     * Peer Node Id. The node id is an unsigned 64 bit integer encoded as a hex string.
```

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 5]
```diff
@@ -245,6 +246,21 @@ public CryptographicParameters getCryptographicParameters(Hash blockHash) throws
         return cryptographicParameters;
     }
 
+    /**
+     * Gets Peer Statistics.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:250]:
```suggestion
     * Gets {@link PeerStatistics} of the node.
```

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Looks good. +1
Only a few minor comments.

Can you add it to the appropriate section of the README.md also?
Here
https://github.com/Concordium/concordium-java-sdk/blob/main/README.md#fill-in
and here
https://github.com/Concordium/concordium-java-sdk/blob/main/README.md#fill-in-1

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java (hunk 5 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/Client.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java (hunk 4 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStatistics.java (hunk 3 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStatistics.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java (hunk 2 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java (hunk 1 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
+package com.concordium.sdk.responses.peerStats;
+
+import concordium.ConcordiumP2PRpc;
+import lombok.Builder;
+import lombok.Getter;
+
+@Builder
+@Getter
+public class PeerStat {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java:9]: Please add a `toString` for convenience.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStatistics.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import concordium.ConcordiumP2PRpc;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.val;
+
+import java.util.stream.Collectors;
+
+@Builder
+@Getter
+public class PeerStatistics {
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStatistics.java:13]: Please add a `toString` for convenience.

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Looks good. 
Could you please add `toString` for `PeerStat` and `PeerStatistics` for convenience.

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java (hunk 6 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStat.java:11]: done

File: concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStatistics.java (hunk 7 above)
Comment (parv0888) [reply concordium-sdk/src/main/java/com/concordium/sdk/responses/peerStats/PeerStatistics.java:15]: done

File: README.md [hunk 8]
```diff
@@ -195,6 +195,12 @@ long getTotalSent()
 ```
 Retrives total number of packets sent from the node.
 
+- `getPeerStatistics`
+```java
+PeerStatistics getPeerStatistics(final boolean includeBootstrappers)
+```
+Retrieves the `PeerStatistics` the node is connected to.
```
Comment (EmilLa1) [inline README.md:202]:
```suggestion
Retrieves the statistics of the peers that the node is connected to.
```

File: concordium-sdk/src/main/java/com/concordium/sdk/Client.java [hunk 9]
```diff
@@ -263,6 +264,21 @@ public long getTotalSent() {
         return server().peerTotalSent(ConcordiumP2PRpc.Empty.newBuilder().build()).getValue();
     }
 
+    /**
+     * Gets {@link PeerStatistics} of the node.
+     * @param includeBootstrappers Should include Bootstrappers in the response.
```
Comment (EmilLa1) [inline concordium-sdk/src/main/java/com/concordium/sdk/Client.java:269]:
```suggestion
     * @param includeBootstrappers Whether bootstrappers should be included in the response.
```

**Review by EmilLa1 — CHANGES_REQUESTED** [review]
Looks good! 
I suggest minor rephrasings a few places and it's good to go. 👍

**Review by EmilLa1 — APPROVED** [review]
Looks good 👍 

Please rebase and it's ready for merge.

**Pull request merged**